electrolytes/solvation_shell_utils.py

```diff
@@ -0,0 +1,157 @@
+        shell for shell in shells if not shell.intersection(solute_atoms)
+    ]
+    return [st.extract(shell, copy_props=True) for shell in filtered_shells]
+
+
+def expand_shell(
+    st: Structure,
+    shell_ats: Set[int],
+    central_solute: int,
+    radius: float,
+    solute_res_names: List[str],
+    max_shell_size: int = 200,
+) -> Structure:
+    """
+    Expands a solvation shell. If there are any (non-central) solutes present in the shell,
+    recursively include shells around those solutes.
+    First, gets the molecule numbers of solute molecules that are within the radius
+    and not already expanded around. Then, continuously expand around them as long as we don't hit an atom limit.
+    Args:
+        st: Entire structure from the PDB file
+        shell_ats: Set of atom indices (of `st`) in a shell (1-indexed)
+        central_solute: Molecule index (of 'st') of the central solute in the shell
+        radius: Solvation radius (Angstroms) to consider
+        solute_res_names: List of residue names that correspond to solute atoms in the simulation
+        max_shell_size: Maximum size (in atoms) of the expanded shell
+    Returns:
+        Structure object containing the expanded solvation shell
+    """
+    # # TODO: I'm not sure why we have to do this, but sometimes the central solute is not in the shell
+    # shell_ats.add(st.molecule[central_solute].getAtomList())
+    solutes_included = set([central_solute])
+
+    def get_new_solutes(st, shell_ats, solutes_included, solute_res_names):
+        new_solutes = set()
+        for at in shell_ats:
+            # If atom is part of a non-central solute molecule - should expand the shell
+            if (
+                st.atom[at].molecule_number not in solutes_included
+                and st.atom[at].getResidue().pdbres.strip() in solute_res_names
+            ):
+                new_solutes.add(st.atom[at].molecule_number)
+        return new_solutes
+
+    new_solutes = get_new_solutes(st, shell_ats, solutes_included, solute_res_names)
+    while new_solutes:
+        # add entire residues within solvation shell radius of any extra solute atoms
+        new_shell_ats = shell_ats.union(
+            analyze.evaluate_asl(
+                st,
+                f'fillres within {radius} mol {",".join([str(i) for i in new_solutes])}',
+            )
+        )
+        if len(new_shell_ats) <= max_shell_size:
+            shell_ats = new_shell_ats
+            solutes_included.update(new_solutes)
+            new_solutes = get_new_solutes(
+                st, shell_ats, solutes_included, solute_res_names
+            )
+        else:
+            break
+
+    shell_ats = sorted(shell_ats)
+    final_structure = st.extract(shell_ats, copy_props=True)
+
+    # find index of first atom of the central solute in the sorted shell_ats (adjust for 1-indexing)
+    central_solute_atom_idx = (
+        shell_ats.index(st.molecule[central_solute].getAtomList()[0]) + 1
+    )
```

Comment:
```suggestion
    central_at = st.molecule[central_solute].atom[1]
    central_at.property['b_m_central'] = True
    final_structure = st.extract(shell_ats, copy_props=True)

    # find index of first atom of the central solute in the sorted shell_ats (adjust for 1-indexing)
    central_solute_atom_idx = next(at for at in final_structure.atom if at.property.pop('b_m_central', False)).index
```

That probably does work, but we can do it a little more robustly by just labeling that atom in the original structure and checking the label in the extracted structure (note that Schrodinger properties have to be of the form X_Y_Z where X=b,s,i,r for bool, string, int, or real, Y and Z can be anything)